### PR TITLE
Cleanup azure-client-sdk-parent Configuration

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR centralized and de-duplicates configurations defined in `azure-client-sdk-parent`.

High-level overview of most changes were moving configurations into `pluginManagement`, having `plugins` configure executions that are expected to run in all children POMs, and remove configurations from `reporting` that will be inherited from `pluginManagement`. Smaller changes were moving common configurations from profiles into the top-level configuration.

### CODE REVIEW

File: sdk/parents/azure-arm-parent/pom.xml [hunk 1]
```diff
@@ -672,7 +672,7 @@
         <configuration>
           <doctitle>Microsoft Azure Management Libraries for Java ${project.version}</doctitle>
           <windowtitle>Microsoft Azure Management Libraries for Java ${project.version}</windowtitle>
-          <additionalparam>-Xdoclint:none</additionalparam>
+          <doclint>false</doclint>
```
Comment (srnagar) [inline sdk/parents/azure-arm-parent/pom.xml:675]: `<additionalparam>-Xdoclint:none</additionalparam>`	and `<doclint>false</doclint>` are essentially the same right?

File: pom.xml [hunk 2]
```diff
@@ -381,7 +381,7 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-checkstyle-plugin</artifactId>
-            <version>3.1.0</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
+            <version>3.1.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
```
Comment (srnagar) [inline pom.xml:384]: Don't we need to update this version in `external_dependencies.txt`?

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 3]
```diff
@@ -110,46 +110,26 @@
...
-          <failOnWarning>true</failOnWarning>
-        </configuration>
       </plugin>
 
       <!-- This plugin scans checkstyle issues in the code -->
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
         <version>3.1.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
-        <dependencies>
```
Comment (srnagar) [inline sdk/parents/azure-client-sdk-parent/pom.xml:129]: There's a `<reporting>` section that can probably be cleaned up as well. Also, the javadoc section with all the groups can be removed too.

File: sdk/parents/azure-arm-parent/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/parents/azure-arm-parent/pom.xml:675]: Yes and `additionalparam` isn't a valid configuration option for the Javadoc plugin. https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#doclint

File: pom.xml (hunk 2 above)
Comment (alzimmermsft) [reply pom.xml:384]: External dependencies was already updated and from what I remember correctly this file doesn't get validated for having the correct version tags.

File: sdk/parents/azure-client-sdk-parent/pom.xml (hunk 3 above)
Comment (alzimmermsft) [reply sdk/parents/azure-client-sdk-parent/pom.xml:129]: Removed all aggregate reporting configurations, leaving the non-aggregate configurations for now as children POMs inherit it. We can have a follow-up investigation on whether those should be removed as well.

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 4]
```diff
@@ -469,6 +388,112 @@
...
+            <includeTests>false</includeTests>
+
+            <!-- Enable this plugin when issues documented are fixed -->
+            <!--          <plugins>-->
+            <!--            <plugin>-->
+            <!--              <groupId>com.h3xstream.findsecbugs</groupId>-->
+            <!--              <artifactId>findsecbugs-plugin</artifactId>-->
+            <!--              <version>1.9.0</version> &lt;!&ndash; {x-version-update;com.h3xstream.findsecbugs:findsecbugs-plugin;external_dependency} &ndash;&gt;-->
+            <!--            </plugin>-->
+            <!--          </plugins>-->
```
Comment (srnagar) [inline sdk/parents/azure-client-sdk-parent/pom.xml:492]: What are the issues that need to be fixed to re-enable this? This seems to be commented out for a long time. Should we re-enable this and see what issues crop up and file issues to track them?

**Review by srnagar — APPROVED** [review]

File: sdk/parents/azure-client-sdk-parent/pom.xml (hunk 4 above)
Comment (alzimmermsft) [reply sdk/parents/azure-client-sdk-parent/pom.xml:492]: @g2vinay do you know if these have been resolved?

I can file a follow-up issue to enable this.

**Pull request merged**